packages/@aws-cdk/aws-appmesh/lib/virtual-gateway-listener.ts

```diff
@@ -32,6 +39,46 @@ export interface HttpGatewayListenerOptions {
    * @default - none
    */
   readonly tlsCertificate?: TlsCertificate;
+
+  /**
+   * Connection pool for http listeners
+   *
+   * @default - None
+   */
+  readonly connectionPool?: HttpConnectionPool;
+}
+
+/**
+ * Represents the properties needed to define HTTP2 Listeners for a VirtualGateway
+ */
+export interface Http2GatewayListenerOptions {
+  /**
+   * Port to listen for connections on
+   *
+   * @default - 8080
+   */
+  readonly port?: number
+
+  /**
+   * The health check information for the listener
+   *
+   * @default - no healthcheck
+   */
+  readonly healthCheck?: HealthCheck;
+
+  /**
+   * Represents the configuration for enabling TLS on a listener
+   *
+   * @default - none
+   */
+  readonly tlsCertificate?: TlsCertificate;
```

Comment:
Can we introduce a new, module-private, interface that both `HttpGatewayListenerOptions` and `Http2GatewayListenerOptions` extend, that contains all of these properties, so there is no duplication?